### PR TITLE
Add basic counter support to the typescript SDK.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @damccorm

**Review by damccorm — APPROVED** [review]
Mostly LGTM, just had one question about non-updated metrics

File: sdks/typescript/src/apache_beam/worker/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    } else {
+      this.count += 1;
+      this.sum += value;
+      this.min = Math.min(value, this.min);
+      this.max = Math.max(value, this.max);
+    }
+  }
+
+  reset(): void {
+    this.count = 0;
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/typescript/src/apache_beam/worker/metrics.ts:122]: Ah, good catch. Nothing should be done with a metric that hasn't been updated at least once. I've added some logic to only actually report those counters that are actually updated. In addition, as a precaution, I've also updated the distribution merge function.

[the remark this answers is not in the document]
Comment (damccorm) [reply sdks/typescript/src/apache_beam/worker/metrics.ts:122]: Sounds good, this LGTM (other than the new merge conflicts)

**Pull request merged**